server/events/vcs/pull_status_fetcher.go

```diff
@@ -5,29 +5,33 @@ import (
 	"github.com/runatlantis/atlantis/server/events/models"
 )
 
+//go:generate pegomock generate -m --package mocks -o mocks/mock_pull_req_status_fetcher.go PullReqStatusFetcher
+
 type PullReqStatusFetcher interface {
-	FetchPullStatus(repo models.Repo, pull models.PullRequest, vcsstatusname string) (models.PullReqStatus, error)
+	FetchPullStatus(pull models.PullRequest) (models.PullReqStatus, error)
 }
 
 type pullReqStatusFetcher struct {
-	client Client
+	client        Client
+	vcsStatusName string
 }
 
-func NewPullReqStatusFetcher(client Client) PullReqStatusFetcher {
+func NewPullReqStatusFetcher(client Client, vcsStatusName string) PullReqStatusFetcher {
 	return &pullReqStatusFetcher{
-		client: client,
+		client:        client,
+		vcsStatusName: vcsStatusName,
 	}
 }
 
-func (f *pullReqStatusFetcher) FetchPullStatus(repo models.Repo, pull models.PullRequest, vcsstatusname string) (pullStatus models.PullReqStatus, err error) {
-	approvalStatus, err := f.client.PullIsApproved(repo, pull)
+func (f *pullReqStatusFetcher) FetchPullStatus(pull models.PullRequest) (pullStatus models.PullReqStatus, err error) {
```

Comment:
reduce following parameters

- `repo models.Repo`
  - `pull.BaseRepo` is enough.
- `vcsstatusname string`
  - this is a server configuration, so it should have be in the `pullReqStatusFetcher` struct.